crates/bevy_core_pipeline/src/tonemapping/node.rs

```diff
@@ -2,72 +2,61 @@ use std::sync::Mutex;
 
 use crate::tonemapping::{TonemappingLuts, TonemappingPipeline, ViewTonemappingPipeline};
 
-use bevy_ecs::prelude::*;
-use bevy_ecs::query::QueryState;
+use bevy_ecs::system::lifetimeless::Read;
+use bevy_ecs::{prelude::*, query::ROQueryItem};
 use bevy_render::{
     render_asset::RenderAssets,
-    render_graph::{Node, NodeRunError, RenderGraphContext},
+    render_graph::{NodeRunError, RenderGraphContext, ViewNode},
     render_resource::{
         BindGroup, BindGroupDescriptor, BindGroupEntry, BindingResource, BufferId, LoadOp,
         Operations, PipelineCache, RenderPassColorAttachment, RenderPassDescriptor,
         SamplerDescriptor, TextureViewId,
     },
     renderer::RenderContext,
     texture::Image,
-    view::{ExtractedView, ViewTarget, ViewUniformOffset, ViewUniforms},
+    view::{ViewTarget, ViewUniformOffset, ViewUniforms},
 };
 
 use super::{get_lut_bindings, Tonemapping};
 
 pub struct TonemappingNode {
-    query: QueryState<
-        (
-            &'static ViewUniformOffset,
-            &'static ViewTarget,
-            &'static ViewTonemappingPipeline,
-            &'static Tonemapping,
-        ),
-        With<ExtractedView>,
-    >,
     cached_bind_group: Mutex<Option<(BufferId, TextureViewId, BindGroup)>>,
     last_tonemapping: Mutex<Option<Tonemapping>>,
 }
 
 impl FromWorld for TonemappingNode {
-    fn from_world(world: &mut World) -> Self {
+    fn from_world(_world: &mut World) -> Self {
         Self {
-            query: QueryState::new(world),
             cached_bind_group: Mutex::new(None),
             last_tonemapping: Mutex::new(None),
         }
     }
 }
```

Comment:
Not sure honestly, I think my reasoning at the time was that they wouldn't work with a derive Default so I didn't change them. I'll push an update with the changes.

Reply:
It should work with `derive(Default)` since `Option<T>` implements `Default` regardless of T\[[1]\]. Also `Mutex<T>`  does for `T: Default`\[[2]\]. Also just tested it myself, it does compile.

[1]: https://doc.rust-lang.org/stable/std/option/enum.Option.html#impl-Default-for-Option%3CT%3E
[2]: https://doc.rust-lang.org/stable/std/sync/struct.Mutex.html#impl-Default-for-Mutex%3CT%3E